### PR TITLE
Handle compare() and blend() intrinsics in VectorAPIExpansion

### PR DESCRIPTION
**Pull request by gita-omr**
- add recognized methods for compare and blend intrinsics
- add handlers in VectorAPIExpansion
- reorganize code in VectorAPIExpansion for getting scalar or vector opcodes
since blend() does not have scalar version

### CODE REVIEW

**Comment by gita-omr** [comment]
Fixed typo in the header

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp [hunk 1]
```diff
@@ -1338,7 +1339,9 @@ TR::Node *TR_VectorAPIExpansion::naryIntrinsicHandler(TR_VectorAPIExpansion *opt
    if (elementType == TR::Int8 || elementType == TR::Int16)
       opType = TR::Int32;
 
-   TR::ILOpCodes scalarOpCode = ILOpcodeFromVectorAPIOpcode(vectorAPIOpcode, opType);
+   TR::ILOpCodes scalarOpCode = ILOpcodeFromVectorAPIOpcode(vectorAPIOpcode, opType, true);
```
Comment (knn-k) [inline runtime/compiler/optimizer/VectorAPIExpansion.cpp:1342]: Please add `/*scalar*/` after `true`, like you do at line 1114 above.

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp [hunk 2]
```diff
@@ -1449,37 +1477,98 @@ TR::Node *TR_VectorAPIExpansion::broadcastCoercedIntrinsicHandler(TR_VectorAPIEx
...
+
+      if (!comp->cg()->getSupportsOpCodeForAutoSIMD(vectorOpCode, elementType))
+         return NULL;
+
+      return node;
+      }
+
+   return transformNary(opt, treeTop, node, elementType, vectorLength, mode, scalarOpCode, vectorOpCode,
+                        5/*first operand*/, 2, useVcall);
+
```
Comment (knn-k) [inline runtime/compiler/optimizer/VectorAPIExpansion.cpp:1536]: Please remove this empty line.

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp [hunk 3]
```diff
@@ -1576,7 +1663,9 @@ TR_VectorAPIExpansion::methodTable[] =
    {loadIntrinsicHandler,  TR::NoType, Vector,  {Unknown, elementType, numLanes}},                           // jdk_internal_vm_vector_VectorSupport_load
    {storeIntrinsicHandler, TR::NoType, Unknown, {Unknown, elementType, numLanes, Unknown, Unknown, Vector}}, // jdk_internal_vm_vector_VectorSupport_store
    {binaryIntrinsicHandler,TR::NoType, Vector,  {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector}},  // jdk_internal_vm_vector_VectorSupport_binaryOp
+   {blendIntrinsicHandler,TR::NoType, Vector, {Unknown, Unknown, elementType, numLanes, Vector, Vector, Vector, Unknown}}, // jdk_internal_vm_vector_VectorSupport_blend
```
Comment (knn-k) [inline runtime/compiler/optimizer/VectorAPIExpansion.cpp:1666]: Please add a space before `TR::NoType`.

**Comment by gita-omr** [comment]
Addressed comments above. Thanks @knn-k !

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp (hunk 3 above)
Comment (knn-k) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp]: I intended to make it look like a table using space characters.
```
   {loadIntrinsicHandler,  TR::NoType, Vector,  {Unknown, elementType, numLanes}},                           // jdk_internal_vm_vector_VectorSupport_load
   {storeIntrinsicHandler, TR::NoType, Unknown, {Unknown, elementType, numLanes, Unknown, Unknown, Vector}}, // jdk_internal_vm_vector_VectorSupport_store
   {binaryIntrinsicHandler,TR::NoType, Vector,  {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector}},  // jdk_internal_vm_vector_VectorSupport_binaryOp
   {blendIntrinsicHandler, TR::NoType, Vector,  {Unknown, Unknown, elementType, numLanes, Vector, Vector, Vector, Unknown}}, // jdk_internal_vm_vector_VectorSupport_blend
```

That is already broken by long names of other handlers in the original code, though.
```
   {broadcastCoercedIntrinsicHandler,TR::NoType, Vector, {Unknown, elementType, numLanes, Unknown, Unknown, Unknown}},  // jdk_internal_vm_vector_VectorSupport_broadcastCoerced
   {ternaryIntrinsicHandler,TR::NoType, Vector,  {Unknown, Unknown, Unknown, elementType, numLanes, Vector, Vector}},  // jdk_internal_vm_vector_VectorSupport_ternaryOp
```

**Comment by knn-k** [comment]
jenkins test sanity all jdk18

**Review by knn-k — APPROVED** [review]

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp (hunk 3 above)
Comment (gita-omr) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp]: Sorry @knn-k , missed this comment! I will try to make the table look like that in a next PR.

**Pull request merged**